asserts/findwildcard.go

```diff
@@ -101,11 +117,67 @@ func findWildcardDescend(top, current string, descendantWithWildcard []string, f
 			return fmt.Errorf("findWildcard: invoked with malformed wildcard: %v", err)
 		}
 		if ok {
-			err = findWildcardDescend(top, filepath.Join(current, name), descendantWithWildcard[1:], foundCb)
+			err = findWildcardDescend(top, filepath.Join(current, name), descendantWithWildcard[1:], seqnum, foundCb)
 			if err != nil {
 				return err
 			}
 		}
 	}
 	return nil
 }
+
+func findWildcardSequence(top, current, seqWildcard string, descendantWithWildcard []string, seqnum int, foundCb func(relpath []string) error) error {
+	filter := func(i int) bool { return true }
+	if seqnum != -1 {
+		if seqWildcard == "#>" {
+			filter = func(i int) bool { return i > seqnum }
+		} else {
+			filter = func(i int) bool { return i < seqnum }
+		}
+	}
+
+	d, err := os.Open(current)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+	var seq []int
+	for {
+		names, err := d.Readdirnames(100)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		for _, n := range names {
+			sqn, err := strconv.Atoi(n)
+			if err != nil || sqn < 0 {
+				return fmt.Errorf("cannot parse %q name as a sequence number", filepath.Join(current, n))
+			}
+			if filter(sqn) {
+				seq = append(seq, sqn)
+			}
+		}
+	}
+	sort.Ints(seq)
+
+	var start, dir int
+	if seqWildcard == "#>" {
+		start = 0
+		dir = 1
+	} else {
+		start = len(seq) - 1
+		dir = -1
+	}
+	for i := start; i >= 0 && i < len(seq); i += dir {
+		err = findWildcardDescend(top, filepath.Join(current, strconv.Itoa(seq[i])), descendantWithWildcard, -1, foundCb)
```

Comment:
Although the tree structure is controller, this will trigger interesting behavior if sequence directory name is eg. `"01"`, we'll parse it as `1` and then Itoa() converts it back to `"1"`

Reply:
we'll just miss things but you are right that I should probably review the uses of Atoi in the package (it's an orthogonal problem)

Reply:
Yes, it sounds like a follow up topic if anything.